modules/videoio/src/cap_rtx64gige.cpp

```diff
@@ -0,0 +1,343 @@
+
+        cvInitImageHeader(pIplImageFrameObj, cvSize(pGrabbedFrameObj->Width, pGrabbedFrameObj->Height),
+            pixelDepth, pGrabbedFrameObj->numberOfChannels, IPL_ORIGIN_TL, MAX_BYTES_PER_PIXEL);
+        pIplImageFrameObj->imageData = pGrabbedFrameObj->Image_Data;
+        int widthstep = (pixelDepth / 8) * pGrabbedFrameObj->Width * pGrabbedFrameObj->numberOfChannels;
+        cvSetData(pIplImageFrameObj, pGrabbedFrameObj->Image_Data, widthstep);
+
+        // Setup conversion code to BGR
+        switch (pGrabbedFrameObj->pixelFormat)  // m_PixelFormat
+        {
+        case GVSP_PIX_BAYBG8:
+        case GVSP_PIX_BAYBG10:
+        case GVSP_PIX_BAYBG12:
+        case GVSP_PIX_BAYBG10_PACKED:
+        case GVSP_PIX_BAYBG12_PACKED:
+        case GVSP_PIX_BAYBG16:
+            colorConversionCode = CV_BayerBG2BGR;
+            break;
+
+        case GVSP_PIX_BAYGB8:
+        case GVSP_PIX_BAYGB10:
+        case GVSP_PIX_BAYGB12:
+        case GVSP_PIX_BAYGB10_PACKED:
+        case GVSP_PIX_BAYGB12_PACKED:
+        case GVSP_PIX_BAYGB16:
+            colorConversionCode = CV_BayerGB2BGR;
+            break;
+
+        case  GVSP_PIX_BAYRG8:
+        case  GVSP_PIX_BAYRG10:
+        case  GVSP_PIX_BAYRG12:
+        case  GVSP_PIX_BAYRG10_PACKED:
+        case  GVSP_PIX_BAYRG12_PACKED:
+        case  GVSP_PIX_BAYRG16:
+            colorConversionCode = CV_BayerRG2BGR;
+            break;
+
+        case GVSP_PIX_BAYGR16:
+        case GVSP_PIX_BAYGR12_PACKED:
+        case GVSP_PIX_BAYGR10_PACKED:
+        case GVSP_PIX_BAYGR12:
+        case GVSP_PIX_BAYGR10:
+        case GVSP_PIX_BAYGR8:
+            colorConversionCode = CV_BayerGR2BGR;
+            break;
+
+        default:
+            colorConversionCode = 0;
+        }
+
+        switch (pGrabbedFrameObj->pixelFormat)  // m_PixelFormat
+        {
+        case GVSP_PIX_BAYBG8:
+        case GVSP_PIX_BAYGR8:
+        case GVSP_PIX_BAYGB8:
+        case  GVSP_PIX_BAYRG8:
+            inputArrayType = CV_8UC1;
+            outputArrayType = CV_8UC3;
+            break;
+
+        case GVSP_PIX_BAYBG16:
+        case GVSP_PIX_BAYGB16:
+        case  GVSP_PIX_BAYRG16:
+        case GVSP_PIX_BAYGR16:
+            inputArrayType = CV_16UC1;
+            outputArrayType = CV_16UC3;
+            break;
+
+        default:
+            inputArrayType = 0;
+            outputArrayType = 0;
+        }
+
+        if (colorConversionCode &&  outputArrayType)
+        {
+            // convert bayer to BGR
+            pGrabbedFrameObj->numberOfChannels = IMAGE_NUM_CHANNELS;
+            pGrabbedFrameObj->CvArrayType = outputArrayType;
+
+
+            cv::Mat mImageSrc(pIplImageFrameObj->height, pIplImageFrameObj->width, inputArrayType,
+                pIplImageFrameObj->imageData, pIplImageFrameObj->widthStep);
+            cv::Mat mImageDest(pIplImageFrameObj->height, pIplImageFrameObj->width, outputArrayType);
+
+            cv::cvtColor(mImageSrc, mImageDest, colorConversionCode);
```

Comment:
There is no real handling for `outputArrayType`.

Probably `cv::Mat mImageDest;` is enough. Or use "const Mat" to avoid reallocation.